### ISSUE TITLE
Possible speedup in `random_crop_and_flip`

### ISSUE THREAD

**Issue by goodfeli** [tensorflow/cleverhans#695]
`_pad_img` could be rewritten to operate on a whole batch, and avoid needing a map_fn. Check how this affects runtime of CIFAR-10 training pipeline.

**Comment by david-berthelot** [comment]
Without map_fn, the cropping happens at the same location within a batch. This in turn reduces stochasticity for gradient descent.
